### PR TITLE
chore: add analytics events 

### LINKED ISSUE #27557: [Task]: Activation: add analytics events for new onboarding flow
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Add analytics events described [here](https://www.notion.so/By-improving-the-templates-discoverability-we-ll-increase-the-of-users-who-start-from-a-template--5f219f72c5d74466a31662b9d928b01d). 
- Following events will be covered as part of this issue, rest of the events to be covered via a different issue: 
	- How many users selected start from an app and start from scratch?
	- How many users selected a template?
	- How many previewed a template?
	- How many forked a template?
	- How many users requested new templates?
	  - What are their requests?
	- How many saw the list of templates, clicked on back and started from scratch?

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- Add analytics events for the [new onboarding flow](https://www.notion.so/appsmith/f478fa5016a548c081c8f5e124e74ba9?v=2fc107badb724eb7aee27275748f1c05&p=5f219f72c5d74466a31662b9d928b01d&pm=s) where user can choose to either start form scratch or start from template. 
- The following events are added as part of this PR: 
	- How many users selected start from an app and start from scratch?
	- How many users selected a template?
	    - How many previewed a template?
	    - How many forked a template?
	- How many users requested new templates?
	    - What are their requests?
	- How many saw the list of templates, clicked on back and started from scratch?
- Rest of the events will be added as part of a new PR. 

#### PR fixes following issue(s)
Fixes #27557 

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- Automation test seems not possible at the moment. 

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx [hunk 1]
```diff
@@ -156,17 +159,23 @@ const CreateNewAppsOption = ({
   };
 
   const onTemplateClick = (id: string) => {
+    AnalyticsUtil.logEvent("CLICK_ON_TEMPLATE_CARD_WHEN_ONBOARDING", { id });
```
Comment (sneha122) [inline app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx:162]: @sumitsum I hope template id is enough in the events, and we dont need template name

File: app/client/src/pages/Templates/Template/RequestTemplate.tsx [hunk 2]
```diff
@@ -51,6 +52,7 @@ const REQUEST_TEMPLATE_URL =
 
 function RequestTemplate() {
   const onClick = () => {
+    AnalyticsUtil.logEvent("REQUEST_NEW_TEMPLATE");
```
Comment (sneha122) [inline app/client/src/pages/Templates/Template/RequestTemplate.tsx:55]: @sumitsum Do we not want to capture what requests users have made? What kinda templates they are requesting?

File: app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx [hunk 3]
```diff
@@ -141,6 +142,8 @@ const CreateNewAppsOption = ({
   const isImportingTemplate = useSelector(isImportingTemplateSelector);
   const dispatch = useDispatch();
   const onClickStartFromTemplate = () => {
+    AnalyticsUtil.logEvent("CREATE_APP_FROM_TEMPLATE");
```
Comment (sneha122) [inline app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx:145]: @sumitsum I dont see the events for back button here, but in PR description it is mentioned, can we please check this?

Comment (sumitsum) [reply app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx:147]: Done.

File: app/client/src/pages/Templates/Template/RequestTemplate.tsx (hunk 2 above)
Comment (sumitsum) [reply app/client/src/pages/Templates/Template/RequestTemplate.tsx:55]: Hi @sneha122 we cannot capture it here. The request is made via an appsmith app. Also, Balaji has [mentioned](https://theappsmith.slack.com/archives/C056PSE5X0E/p1695377154758509?thread_ts=1695197879.999289&cid=C056PSE5X0E) that he is ok with it.

File: app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx (hunk 1 above)
Comment (sumitsum) [reply app/client/src/ce/pages/Applications/CreateNewAppsOption.tsx]: Done.

**Review by sneha122 — APPROVED** [review]

**Pull request merged**